MREUnityRuntime/MREUnityRuntimeLib/Core/ActorManager.cs

```diff
@@ -2,19 +2,23 @@
 // Licensed under the MIT License.
 using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using MixedRealityExtension.API;
 using MixedRealityExtension.App;
 using MixedRealityExtension.Core.Interfaces;
 using MixedRealityExtension.IPC;
 using MixedRealityExtension.Messaging;
+using MixedRealityExtension.Messaging.Commands;
+using MixedRealityExtension.Messaging.Payloads;
 
 namespace MixedRealityExtension.Core
 {
-    internal class ActorManager
+    internal class ActorManager : ICommandHandlerContext
     {
         private MixedRealityExtensionApp _app;
         private Dictionary<Guid, Actor> _actorMapping = new Dictionary<Guid, Actor>();
-        private HashSet<Guid> _reservations = new HashSet<Guid>();
+        private Dictionary<Guid, ActorCommandQueue> _actorCommandQueues = new Dictionary<Guid, ActorCommandQueue>();
+        private List<Action> _uponStable = new List<Action>();
```

Comment:
>_uponStable [](start = 29, length = 11)

Note: This is only needed due to the way we sync animations currently. Once that is refactored this can go away. We're going to change it to work like sound does.